core/src/main/java/org/infinispan/partitionhandling/impl/AvailablePartitionHandlingManager.java

```diff
@@ -37,13 +39,13 @@ public void setAvailabilityMode(AvailabilityMode availabilityMode) {/*no-op*/}
    public void checkWrite(Object key) {/*no-op*/}
 
    @Override
-   public void checkRead(Object key) {/*no-op*/}
+   public void checkRead(Object key, Set<Flag> flags) {/*no-op*/}
```

Comment:
Thinking we may want to use `long` instead of `Set<Flag>` as is used in other places now.

Reply:
Then you can check the flag being set via `FlagBitSets` etc.